lib/core/application/value_validator.dart

```diff
@@ -34,3 +34,11 @@ Either<ValueFailure, Unit> validatePostal(String input) {
 
 Either<ValueFailure, Unit> validateNotEmpty(String input) =>
     input != '' ? right(unit) : left(const ValueFailure.empty());
+
+Either<ValueFailure, Unit> validateUsualPrice(double input) {
+  return input >= 0.0 && input <= 10000 ? right(unit) : left(const ValueFailure.invalidPriceValue());
+}
+
+Either<ValueFailure, Unit> validateDiscountedPrice(double input, double usualPrice) {
+  return input >= 0.0 && input <= 10000 && input < usualPrice ? right(unit) : left(const ValueFailure.invalidPriceValue());
+}
```

Comment:
for validation of usual price and discounted price u can check if it is empty first, if it is then return empty failure, then proceed to check with invalid value